### PR TITLE
Highlight Logic Fixes

### PR DESCRIPTION
**Pull request by nirdoshyadav**
Juan, I have made couple of fixes in the existing alogorithm
- If multiple sentences are on the same div, highlight the full div
- Updated the Regex, removed the ' ' after the term so it will also highlight the terms having '.' after the term

### CODE REVIEW

File: web/pdf_find_controller.js [hunk 1]
```diff
@@ -142,59 +142,36 @@ var PDFFindController = (function PDFFindControllerClosure() {
...
-      }*/
-      //Nirdosh: regular expression to match full sentence
-      var regex=new RegExp('[^.]* '+query+' [^.]*\.','gi');
-      var match;
-      while ((match = regex.exec(pageContent)) !== null) {
-        if (match.index === regex.lastIndex) {
-             ++regex.lastIndex;
+
+        //Nirdosh: regular expression to match full sentence
+        var regex=new RegExp("[^.]* "+query+"[^.]*\.","gi");
```
Comment (MASmedios) [inline web/pdf_find_controller.js]: Use single quotes with strings

File: web/pdf_find_controller.js [hunk 2]
```diff
@@ -142,59 +142,36 @@ var PDFFindController = (function PDFFindControllerClosure() {
...
+        var match;
+        // console.log(query);
+        while ((match = regex.exec(pageContent)) != null) {
```
Comment (MASmedios) [inline web/pdf_find_controller.js]: Compare with !== to compare value and type:
http://www.w3schools.com/js/js_comparisons.asp

File: web/pdf_find_controller.js [hunk 3]
```diff
@@ -276,7 +253,7 @@ var PDFFindController = (function PDFFindControllerClosure() {
         page.textLayer.updateMatches();
       }
       //scroll match into view
-      if (PDFJS.multiple !== undefined) {
+      if (PDFJS.multiple == undefined) {
```
Comment (MASmedios) [inline web/pdf_find_controller.js]: Compare with === 
To compare undefine type use typeof:
if (typeof PDFJS.multiple === 'undefined')

File: web/pdf_find_controller.js [hunk 4]
```diff
@@ -326,15 +303,13 @@ var PDFFindController = (function PDFFindControllerClosure() {
               if (PDFJS.multiple === undefined) {
```
Comment (MASmedios) [inline web/pdf_find_controller.js]: Use typeof

File: web/pdf_find_controller.js [hunk 5]
```diff
@@ -326,15 +303,13 @@ var PDFFindController = (function PDFFindControllerClosure() {
...
               } else { 
-                if(page === pageIdx) {
-                  //for(var k=0;k < PDFJS.multiple.length;k++){
-                      for (var j = 0; j < PDFJS.multiple.terms.length; j++) {
-                        qwy = PDFJS.multiple.terms[j].replace(/\=/ig, ' ');
-                        self.state.query = qwy;
-                        self.calcSearchMatch(pageIdx);
+                    for(var j=0;j<PDFJS.multiple.length;j++){
+                     //get the data for the current page
+                     if(PDFJS.multiple[j].page == pageIdx+1){
```
Comment (MASmedios) [inline web/pdf_find_controller.js]: Use ===

File: web/text_layer_builder.js [hunk 6]
```diff
@@ -431,22 +431,50 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
...
 
       function beginText(begin, className) {
+        // console.log('Begin Text');
+        // console.log(begin);
         var divIdx = begin.divIdx;
-        textDivs[divIdx].textContent = '';
-        appendTextToDiv(divIdx, 0, begin.offset, className);
+        var div = textDivs[divIdx];
+        var endIndex=begin.offset;
+        var spans = div.getElementsByTagName("span");
```
Comment (MASmedios) [inline web/text_layer_builder.js]: Use single quote

File: web/text_layer_builder.js [hunk 7]
```diff
@@ -431,22 +431,50 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
...
+        var content=bidiTexts[divIdx].str;
+        if(spans!= 'undefined' && spans.length>0){
```
Comment (MASmedios) [inline web/text_layer_builder.js]: Use !== and typeof

File: web/text_layer_builder.js [hunk 8]
```diff
@@ -431,22 +431,50 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
...
         var content = bidiTexts[divIdx].str.substring(fromOffset, toOffset);
         var node = document.createTextNode(content);
         if (className) {
-          var span = document.createElement('span');
-              span.style.backgroundColor='red';
-          //span.className = className;
-          span.appendChild(node);
-          div.appendChild(span);
-          return;
+          var spans=div.getElementsByTagName("span");
```
Comment (MASmedios) [inline web/text_layer_builder.js]: Use single quote

File: web/text_layer_builder.js [hunk 9]
```diff
@@ -431,22 +431,50 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
...
+          if(spans!= 'undefined' && spans.length>0){
```
Comment (MASmedios) [inline web/text_layer_builder.js]: Use typeof and !==

File: web/text_layer_builder.js [hunk 10]
```diff
@@ -468,15 +496,16 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
         var highlightSuffix = (isSelected ? 'selected' : '');
 
         if (PDFJS.multiple === undefined) {
```
Comment (MASmedios) [inline web/text_layer_builder.js]: Use typeof

**Comment by MASmedios** [comment]
Hi Nirdosh,
the original search feature is broken. When i try to use it only search terms in the same page:
image

Can you take a look on this?

**Comment by nirdoshyadav** [comment]
I am looking into this. This is broken from earlier so you can merge this pull request. Once I find the fix, I will create a new pull request.

Thanks,

**Pull request merged**